File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        // Time Complexity:O(n) , Space Complexity:O(1)
        int mini=prices[0];
        int profit=0;
        for(int i=1;i<prices.length;i++){
            int cost=prices[i]-mini;
            profit=Math.max(profit,cost);
            mini=Math.min(mini,prices[i]);
        }
        return profit;
        
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.ts
function maxProfit(prices: number[]): number {
    let minElem:number = prices[0];
    let profit:number = 0;
    for(let value of prices){
        minElem = Math.min(minElem,value);
        profit = Math.max(profit,value-minElem)
    }
    return profit
};